File: app/src/main/java/com/lucasurbas/counter/app/di/ActivityModule.java
package com.lucasurbas.counter.app.di;

import android.support.annotation.NonNull;

import dagger.Module;
import dagger.Provides;
import com.lucasurbas.counter.ui.BaseActivity;
import com.lucasurbas.counter.app.di.scope.ActivityScope;

@Module
public class ActivityModule {

    private final BaseActivity activity;

    public ActivityModule(@NonNull final BaseActivity activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityScope
    BaseActivity provideActivity() {
        return activity;
    }
}

File: app/src/main/java/com/lucasurbas/counter/app/di/ActivityModule.kt
package com.lucasurbas.counter.app.di

import android.support.v4.app.FragmentActivity

import com.lucasurbas.counter.app.di.scope.ActivityScope

import dagger.Module
import dagger.Provides

@Module
class ActivityModule(private val activity: FragmentActivity) {

    @Provides
    @ActivityScope
    internal fun provideActivity(): FragmentActivity {
        return activity
    }
}
